File: Project/MyProject/MyProject/Source/MyProject/Private/BasicFrame/Libs/Core/Object/GPoolObject.cpp
#include "MyProject.h"
#include "GPoolObject.h"
#include "TypeDef.h"
#include "MClassFactory.h"

MY_BEGIN_NAMESPACE(MyNS)

M_IMPLEMENT_AND_REGISTER_CLASS(GPoolObject, MyAllocatedObject)

GPoolObject::GPoolObject()
{
	this->mIsUsePool = false;
	this->onGetFromPool();
}

void GPoolObject::init()
{
	this->onInit();
}

void GPoolObject::onInit()
{

}

void GPoolObject::onGetFromPool()
{

}

void GPoolObject::onPutInPool()
{

}

void GPoolObject::dispose()
{
	this->onPutInPool();
	this->onDestroy();
}

void GPoolObject::destroy()
{
	this->onDestroy();
}

void GPoolObject::onDestroy()
{

}

bool GPoolObject::isUsePool()
{
	return this->mIsUsePool;
}

void GPoolObject::setIsUsePool(bool value)
{
	this->mIsUsePool = value;
}

void GPoolObject::putInPool()
{
	this->onPutInPool();
}

void GPoolObject::getFromPool()
{
	this->onGetFromPool();
}

void GPoolObject::setGlobalUniqueId(std::string value)
{
	this->mGlobalUniqueId = value;
}

std::string GPoolObject::getGlobalUniqueId()
{
	return this->mGlobalUniqueId;
}

void GPoolObject::resetDefault()
{

}

MY_END_NAMESPACE

File: Project/MyProject/Tool/CSToolKit/Project/ToolKit/ToolKit/Source/Libs/Core/GPoolObject.cs
namespace SDK.Lib
{
    /*
     * @brief 基本缓冲对象
     */
    public class GPoolObject : GObjectBase
    {
        protected bool mIsUsePool;      // 是否使用 Pool
        protected int mNumUniqueId;
        protected string mStrUniqueId;

        public GPoolObject()
        {
            this.mIsUsePool = false;
            this.onGetFromPool();
        }

        virtual public void init()
        {
            this.onInit();
        }

        virtual protected void onInit()
        {

        }

        virtual protected void onGetFromPool()
        {

        }

        virtual protected void onPutInPool()
        {

        }

        virtual public void dispose()
        {
            this.onPutInPool();
            this.onDestroy();
        }

        virtual public void destroy()
        {
            this.onDestroy();
        }

        virtual protected void onDestroy()
        {

        }

        public bool isUsePool()
        {
            return this.mIsUsePool;
        }

        public void setIsUsePool(bool value)
        {
            this.mIsUsePool = value;
        }

        public void putInPool()
        {
            this.onPutInPool();
        }

        public void getFromPool()
        {
            this.onGetFromPool();
        }

        public void setNumUniqueId(int value)
        {
            this.mNumUniqueId = value;
        }

        public int getNumUniqueId()
        {
            return this.mNumUniqueId;
        }

        public void setStrUniqueId(string value)
        {
            this.mStrUniqueId = value;
        }

        public string getStrUniqueId()
        {
            return this.mStrUniqueId;
        }
    }
}